### PR TITLE
Sharee substring

### LINKED ISSUE #547: Searching sharee with displayname
Only posible to get sharees by using the username:

Username: `einstein`
Display Name: `Albert Einstein`

results are fetched only matching `einstein`
`Albert`, `Einstein`, `Ein`, `Albe` do not match and response is empty (always forgetting the case sensitiveness)

oC10: matches with both.

Backend: EOS storage via via https://github.com/owncloud-docker/compose-playground/blob/master/examples/hetzner-deploy/make_ocis_eos_compose_test.sh
Version: v1.0.0-rc1
Client: iOS App v11.4 over iOS13

CC @pmaier1

### PR DESCRIPTION
**Pull request by rhafer**
## Description
Allow configuration of substring search for searching sharees
    
This introduces new settings for the users and groups services. "group_substring_filter_type" for the group services and
"user_substring_filter_type" for the users service. They allow to set the type of LDAP filter that is used for substring user searches. Possible values are: "initial", "final" and "any" to do either prefix, suffix or full substring searches.

## Related Issue
- Fixes #547

### CODE REVIEW

**Comment by micbar** [comment]
Awesome!

@mmattel Can you look at it with the 👀 of a strong ENV variable naming and description master? 😀

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-7</strong>](https://drone.owncloud.com/owncloud/ocis/13624/48/1) failed. Further test are cancelled...

File: services/groups/pkg/config/config.go [hunk 1]
```diff
@@ -66,22 +66,23 @@ type Drivers struct {
...
-	UserObjectClass  string          `yaml:"user_object_class" env:"LDAP_USER_OBJECTCLASS;GROUPS_LDAP_USER_OBJECTCLASS" desc:"The object class to use for users in the default user search filter ('inetOrgPerson')."`
-	GroupObjectClass string          `yaml:"group_object_class" env:"LDAP_GROUP_OBJECTCLASS;GROUPS_LDAP_GROUP_OBJECTCLASS" desc:"The object class to use for groups in the default group search filter ('groupOfNames')."`
-	IDP              string          `yaml:"idp" env:"OCIS_URL;OCIS_OIDC_ISSUER;GROUPS_IDP_URL" desc:"The identity provider value to set in the group IDs of the CS3 group objects for groups returned by this group provider."`
-	UserSchema       LDAPUserSchema  `yaml:"user_schema"`
-	GroupSchema      LDAPGroupSchema `yaml:"group_schema"`
+	URI                      string          `yaml:"uri" env:"LDAP_URI;GROUPS_LDAP_URI" desc:"URI of the LDAP Server to connect to. Supported URI schemes are 'ldaps://' and 'ldap://'"`
+	CACert                   string          `yaml:"ca_cert" env:"LDAP_CACERT;GROUPS_LDAP_CACERT" desc:"Path to a CA certificate file for validating the LDAP server's TLS certificate. If empty the system default CA bundle will be used."`
```
Comment (mmattel) [inline services/groups/pkg/config/config.go:70]:
```suggestion
	CACert                   string          `yaml:"ca_cert" env:"LDAP_CACERT;GROUPS_LDAP_CACERT" desc:"Path to a CA certificate file for validating the LDAP server's TLS certificate. If empty, the system default CA bundle will be used."`
```

File: services/users/pkg/config/config.go [hunk 2]
```diff
@@ -69,22 +69,23 @@ type JSONDriver struct {
...
-	UserObjectClass  string          `yaml:"user_object_class" env:"LDAP_USER_OBJECTCLASS;USERS_LDAP_USER_OBJECTCLASS" desc:"The object class to use for users in the default user search filter like 'inetOrgPerson'."`
-	GroupObjectClass string          `yaml:"group_object_class" env:"LDAP_GROUP_OBJECTCLASS;USERS_LDAP_GROUP_OBJECTCLASS" desc:"The object class to use for groups in the default group search filter like 'groupOfNames'. "`
-	IDP              string          `yaml:"idp" env:"OCIS_URL;OCIS_OIDC_ISSUER;USERS_IDP_URL" desc:"The identity provider value to set in the userids of the CS3 user objects for users returned by this user provider."`
-	UserSchema       LDAPUserSchema  `yaml:"user_schema"`
-	GroupSchema      LDAPGroupSchema `yaml:"group_schema"`
+	URI                     string          `yaml:"uri" env:"LDAP_URI;USERS_LDAP_URI" desc:"URI of the LDAP Server to connect to. Supported URI schemes are 'ldaps://' and 'ldap://'"`
+	CACert                  string          `yaml:"ca_cert" env:"LDAP_CACERT;USERS_LDAP_CACERT" desc:"Path to a CA certificate file for validating the LDAP server's TLS certificate. If empty the system default CA bundle will be used."`
```
Comment (mmattel) [inline services/users/pkg/config/config.go:73]:
```suggestion
	CACert                  string          `yaml:"ca_cert" env:"LDAP_CACERT;USERS_LDAP_CACERT" desc:"Path to a CA certificate file for validating the LDAP server's TLS certificate. If empty, the system default CA bundle will be used."`
```

**Comment by mmattel** [comment]
> Can you look at it...

Done, only minor things found

File: services/groups/pkg/config/config.go (hunk 1 above)
Comment (rhafer) [reply services/groups/pkg/config/config.go]: done

**Review by micbar — APPROVED** [review]
LGTM

**Pull request merged**